src/actions/index.js

```diff
@@ -15,35 +14,68 @@ export function receiveLocation(newLocation) {
   };
 }
 
+export function displaySuccess() {
+  return {
+    type: DISPLAY_SUCCESS,
+    message: 'Request succeeded',
+    color: 'green',
+  };
+}
+
+export function displayError(message) {
+  return {
+    type: DISPLAY_ERROR,
+    message,
+    color: 'red',
+  };
+}
+
+export function sendMessage() {
+  return {
+    type: SENDING_MESSAGE,
+    message: 'Sending request...',
+    color: 'black',
+  };
+}
+
 export function fetchLocation(location) {
   let name;
 
   return dispatch => {
+    dispatch(sendMessage());
 
     return geocodeGoogle(location)
       .then(res => {
-        name = res.results[0].formatted_address;
+        if (res.status === 'OK') {
+          name = res.results[0].formatted_address;
 
-        const lat = res.results[0].geometry.location.lat;
-        const lng = res.results[0].geometry.location.lng;
+          const lat = res.results[0].geometry.location.lat;
+          const lng = res.results[0].geometry.location.lng;
```

Comment:
I would advise burying the details of the response in the `geocodeGoogle` method so the knowledge of the API's response format isn't leaking into your actions.